python/lsst/ap/pipe/ap_pipe.py

```diff
@@ -45,160 +57,207 @@
+    defect_tarfile_path = glob(os.path.join(defect_location, DEFECT_TARBALL))[0]
     defectfiles = tarfile.open(defect_tarfile_path).getnames()
-    defectfiles = [os.path.join(defectloc, file) for file in defectfiles]
+    defectfiles = [os.path.join(defect_location, file) for file in defectfiles]
+    return defectfiles
 
-    # Provide location of ref_cats directory containing gaia and pan-starrs tarballs
-    ref_cats = os.path.join(args.dataset, REFCATS_DIR)
 
-    # TEMPORARY HARDWIRED THINGS ARE TEMPORARY
-    # TODO (DM-11422):
-    # - use a coadd as a template instead of a visit
-    # - implement the --id argument and pull visit and ccdnum from there (default = all CCDs)
-    visit = '410985'  # one arbitrary g-band visit in Blind15A40
-    templatevisit = '410929'  # another arbitrary g-band visit in Blind15A40
-    ccdnum = '25'  # arbitrary single CCD for testing
-    sciencevisit = visit  # for doDiffIm, for now
+def get_output_repos(output_root, output_dir):
+    '''
+    Define locations on disk for all of the output repositories used by ap_pipe.
 
-    # Collect useful things to pass to main
-    repolist = [repo, calib_repo, processed_repo, diffim_repo]
-    filelist = [datafiles, calibdatafiles, defectfiles]
-    idlist = [visit, sciencevisit, templatevisit, ccdnum]
+    Parameters
+    ----------
+    output_root: `str`
+        The top-level directory where the output will live.
+    output_dir: `str`
+        Name of directory to be created in output_root.
 
-    return repolist, filelist, idlist, ref_cats
+    Returns
+    -------
+    result: `str`
+        Repository (directory on disk) where desired output product will live.
+    '''
+    if not os.path.isdir(output_root):
+        os.mkdir(output_root)
+    result = os.path.join(output_root, output_dir)
+    return result
 
 
-def main():
+def parsePipelineArgs():
     '''
-    Run each step of the pipeline. NOT used by ap_verify.
+    Parse command-line arguments to run the pipeline. NOT used by ap_verify.
 
-    This main function is solely for the purpose of running ap_pipe alone,
-    from the command line, on a dataset intended for ap_verify. It is useful
-    for testing or standalone image processing independently from verification.
+    Returns
+    -------
+    repos_and_files: `dict`
+        Includes the names of new repos that will be written to disk
+        following ingestion, calib ingestion, processing, and difference imaging
+        (repo, calib_repo, processed_repo, diffim_repo)
+        Includes the files in dataset_root for raw images, flats and biases,
+        and defects (datafiles, calibdatafiles, defectfiles)
+    idlist: `list` containing two `str`
+        Data ID and template info needed for processing and difference imaging
+        [dataId, template]
+        TODO: allow 'template' to be either a visit ID or a repo name (DM-11422)
+    refcats: `str`
+        Path on disk of the reference catalogs
```

Comment:
Why are `idlist` and `refcats` separate from the other results?

Reply:
Primarily because I think about them as different entities. I'll go ahead and add `refcats` to `repos_and_files`, but I'm not going to touch `dataId` and `template` right now because so much around their handling is likely to change in DM-11422.